osu.Game/Screens/Play/SaveFailedScoreButton.cs

```diff
@@ -0,0 +1,141 @@
+        {
+            Size = new Vector2(50, 30);
+            SaveReplay = sr;
+        }
+
+        [BackgroundDependencyLoader(true)]
+        private void load(OsuGame game)
+        {
+            InternalChild = shakeContainer = new ShakeContainer
+            {
+                RelativeSizeAxes = Axes.Both,
+                Child = button = new DownloadButton
+                {
+                    RelativeSizeAxes = Axes.Both,
+                }
+            };
+
+            button.Action = () =>
+            {
+                switch (State.Value)
+                {
+                    case ImportState.Imported:
+                        game?.PresentScore(score, ScorePresentType.Gameplay);
+                        break;
+
+                    case ImportState.Importing:
+                        shakeContainer.Shake();
+                        break;
+
+                    default:
+                        saveScore();
+                        break;
+                }
+            };
+            State.BindValueChanged(updateTooltip, true);
+            State.BindValueChanged(state =>
+            {
+                switch (state.NewValue)
+                {
+                    case ImportState.Imported:
+                        button.State.Value = DownloadState.LocallyAvailable;
+                        break;
+
+                    case ImportState.Importing:
+                        button.State.Value = DownloadState.Importing;
+                        break;
+
+                    case ImportState.Failed:
+                        button.State.Value = DownloadState.NotDownloaded;
+                        break;
+                }
+            }, true);
+        }
+
+        private void saveScore()
+        {
+            State.Value = ImportState.Importing;
+            saveReplayAsync = Task.Run(SaveReplay);
+
+            saveScoreDelegate = new ScheduledDelegate(() =>
+            {
+                if (saveReplayAsync?.IsCompleted != true)
+                    // If the asynchronous preparation has not completed, keep repeating this delegate.
+                    return;
+
+                saveScoreDelegate?.Cancel();
+
+                score = saveReplayAsync.GetAwaiter().GetResult();
+
+                State.Value = score != null ? ImportState.Imported : ImportState.Failed;
+            }, Time.Current, 50);
+
+            Scheduler.Add(saveScoreDelegate);
+        }
+
+        private void updateTooltip(ValueChangedEvent<ImportState> state)
+        {
+            switch (state.NewValue)
+            {
+                case ImportState.Imported:
+                    button.TooltipText = @"Watch replay";
+                    break;
+
+                case ImportState.Importing:
+                    button.TooltipText = @"Importing score";
+                    break;
+
+                case ImportState.Failed:
+                    button.State.Value = DownloadState.NotDownloaded;
```

Comment:
```suggestion
```
Forget to delete,
If everything is ok I will delete it

Reply:
I don't understand. If you forgot, could you please commit a fix for it?

Reply:
Please push the changes. We can't review incomplete changes.